### PR TITLE
Make gen api keys separate job so terraform errors stop pipeline, convert code to use fqdn instead of ip

### PR DESCRIPTION
**Pull request by pritchyspritch**
I also found a bug in the process where we were setting the base url to the url that was already in ssm so the next time a new ec2 spun up the pipeline would have broken.

Going to leave this as draft until we merge #25

### CODE REVIEW

File: generate_api_keys.py [hunk 1]
```diff
@@ -31,6 +32,12 @@ def get_ec2_param(param):
     )["Reservations"][0]["Instances"][0][f"{param}"]
 
 
+def get_fqdn():
+    default_domain = "nessus.gds-cyber-security.digital"
```
Comment (akinnane) [inline generate_api_keys.py:36]: Can you move this into an SSM parameter, please?

File: concourse/pipeline.yml [hunk 2]
```diff
@@ -159,14 +157,58 @@ jobs:
...
+              - -c
+              - |
+                pip install pipenv
+
+                cd cyber-security-nessus-git
+                pipenv --python 3.7
+                pipenv install
+
+                make zip
+                source /usr/local/bin/sts-assume-role.sh 'arn:aws:iam::676218256630:role/nessus_role' 'eu-west-2'
```
Comment (akinnane) [inline concourse/pipeline.yml:188]: Can you move the ARN into a SSM parameter, please?

File: concourse/pipeline.yml [hunk 3]
```diff
@@ -159,14 +157,58 @@ jobs:
...
+                cd terraform
+                terraform init
+                export fqdn=$(terraform output fqdn)
```
Comment (akinnane) [inline concourse/pipeline.yml:191]: Which is the source of truth for the FQDN? This? Line 36 in `generate_api_keys.py`? The `public_base_url` parameter?

Comment (pritchyspritch) [reply concourse/pipeline.yml:191]: My thinking was if someone changes the domain name in our terraform, we will need to update the SSM param we add on line 36. This line 191 updates the fqdn every time we run the terraform so it's always correct. We could add a check to make sure the ssm param is the same and update it if it isn't

Comment (akinnane) [reply concourse/pipeline.yml:191]: I see how it links together now. Thanks for the explanation.

File: generate_api_keys.py (hunk 1 above)
Comment (akinnane) [reply generate_api_keys.py:36]: I don't think we should have a default value for this. If I run this locally without having `fqdn` set it is going to overwrite the SSM parameter with a default value. I think I'd mess up with that and overwrite the SSM value.

**Pull request merged**